### PR TITLE
De-duplicate ManifestFromURI code

### PR DESCRIPTION
**Pull request by petertseng**
You may think it's not so bad that it's duplicated in two places but I'm
about to add a third, so I think it helps to do this.

### CODE REVIEW

**Comment by anthonybishopric** [comment]
:+1:

File: bin/p2-launch/main.go [hunk 1]
```diff
@@ -19,19 +16,10 @@ var (
...
 
-	err = uri.URICopy(*manifestURI, localMan.Name())
-	if err != nil {
-		log.Fatalf("Could not fetch manifest: %s", err)
-	}
-	manifest, err := pods.ManifestFromPath(localMan.Name())
+	manifest, err := pods.ManifestFromURI(*manifestURI)
 	if err != nil {
-		log.Fatalf("Invalid manifest: %s", err)
+		log.Fatalf("%s", err)
```
Comment (bwester) [inline bin/p2-launch/main.go]: For next time: `log.Fatalln(err)`

File: pkg/pods/manifest.go [hunk 2]
```diff
@@ -102,6 +103,21 @@ func ManifestFromBytes(bytes []byte) (*Manifest, error) {
 	return manifest, nil
 }
 
+func ManifestFromURI(manifestUri string) (*Manifest, error) {
+	// Fetch manifest (could be URI) into temp file
+	localMan, err := ioutil.TempFile("", "tempmanifest")
```
Comment (bwester) [inline pkg/pods/manifest.go]: I never noticed this before.

I wonder how many `tempmanifest`-s my laptop has by now...

Comment (petertseng) [inline pkg/pods/manifest.go]: hmm, well I think they'd only be left lying around if the `defer os.Remove` failed to run, right?

**Pull request merged**